gammapy/background/reflected.py

```diff
@@ -89,14 +153,54 @@ def _compute_xy(self, offset, angle):
         y = self.fov['y'] + dy
         return x, y
 
-    def write_off_regions(self, filename):
-        fmt = 'galactic; circle({x},{y},{r})\n'
-        with open(filename, 'w') as fh:
-            for region in self.regions:
-                line = fmt.format(**region)
-                fh.write(line)
 
+def find_reflected_regions_for_event_list(event_list, exclusion,
+                                          on_radius, target=None):
+    """Find reflected regions for a given event list
+
+    Parameters
+    ----------
+    event_list : `~gammapy.data.EventList`
+        Event List
+    exclusion : ImageHDU
+        Excluded regions mask
+    on_radius : `~astropy.coordinates.Angle`
+        Size of the On region
+    target : `~astropy.coordinates.SkyCoord`, optional
+        Source position, taken from event list header if not specified
+    
+    Returns
+    -------
+    off_regions : `~gammapy.background.CircularOffRegions`
+        Table containing the Off regions
+    
+    Examples
+    --------
+    .. code-block:: python
+
+        from gammapy.datasets import gammapy_extra
+        from gammapy.data import EventList 
+        from astropy.io import fits
+
+        exclfile = gammapy_extra.filename('
+             test_datasets/spectrum/dummy_exclusion.fits')
+        eventfile = gammapy_extra.filename('
+             test_datasets/unbundled/hess/run_0023037_hard_eventlist.fits.gz')
+
+        exclusion = fits.open(exclfile, hdu = 0)[0]
+        event_list = EventList.read(eventfile, hdu='EVENTS') 
+        on_rad = Angle(0.3, 'deg')
+
+        regions = find_reflected_regions_for_event_list(
```

Comment:
The event list already has `target_radec` and `pointing_radec` attributes that you could use here (if they work ... probably I just added those without testing).